src/wmtk/PolygonMesh.cpp

```diff
@@ -0,0 +1,239 @@
+#include "PolygonMesh.hpp"
+
+namespace wmtk {
+PolygonMesh::PolygonMesh()
+    : Mesh(2)
+    , m_en_handle(register_attribute<long>("m_en", PrimitiveType::Edge, 2))
+    , m_ep_handle(register_attribute<long>("m_ep", PrimitiveType::Edge, 2))
+    , m_ev_handle(register_attribute<long>("m_ev", PrimitiveType::Edge, 2))
+    , m_ef_handle(register_attribute<long>("m_ef", PrimitiveType::Edge, 2))
+    , m_fh_handle(register_attribute<long>("m_fh", PrimitiveType::Face, 1))
+    , m_vh_handle(register_attribute<long>("m_vh", PrimitiveType::Vertex, 1))
+    , m_f_is_hole_handle(register_attribute<bool>("m_f_is_hole", PrimitiveType::Face, 1))
+{}
+PolygonMesh::PolygonMesh(const PolygonMesh& o) = default;
+PolygonMesh::PolygonMesh(PolygonMesh&& o) = default;
+PolygonMesh& PolygonMesh::operator=(const PolygonMesh& o) = default;
+PolygonMesh& PolygonMesh::operator=(PolygonMesh&& o) = default;
+
+Tuple PolygonMesh::split_edge(const Tuple& t, Accessor<long>& hash_accessor)
+{
+    // TODO Need to implement
+    assert(false);
+    return t;
+}
+
+Tuple PolygonMesh::collapse_edge(const Tuple& t, Accessor<long>& hash_accessor)
+{
+    // TODO Need to implement
+    assert(false);
+    return t;
+}
+
+Tuple PolygonMesh::switch_tuple(const Tuple& tuple, PrimitiveType type) const
+{
+    long new_local_vid = (tuple.m_local_vid + 1) % 2; // All switches swap local vid
+
+    switch (type) {
+    case PrimitiveType::Vertex:
+        return Tuple(
+            new_local_vid,
+            tuple.m_local_eid,
+            tuple.m_local_fid,
+            tuple.m_global_cid,
+            tuple.m_hash);
+
+    case PrimitiveType::Edge: {
+        long h = tuple.m_global_cid;
+        long edge_hid = h % 2;
+
+        // Switched edge is prev for local vertex index 0 and next for local index 1
+        auto e_switch_handle = (tuple.m_local_vid == 0) ? m_ep_handle : m_en_handle;
```

Comment:
could you comment what these handles mean? the other mesh classes should have them too :(.

Reply:
(in the PolygonMesh header presumabaly)